core/trino-parser/src/main/java/io/trino/sql/tree/BoundedQuantifier.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class BoundedQuantifier
```

Comment:
At first, there was only one. Then I decided to split it in two: one for analysis, and another for planner.
In fact, `RangeQuantifier` shouldn't be an AST node, but an IR object. However, I made it an AST node because it is part of the pattern which is all AST.

If I add a parallel IR for patterns, like you suggested in another comment, `RangeQuantifier` will be included in it.

Reply:
This is resolved after adding IR for row pattern.